lib/exception-logger.js

```diff
@@ -1,109 +1,30 @@
-var WebSocket = require('faye-websocket'),
-	Promise = require('promise'),
-	serviceName = "awgproxy",
-	ipAddressingIndex = 0, //for ipv4, 1 for ipv6
-	serviceTimeout = 2000,
-	mdns = require('mdns'),
-	browser = mdns.createBrowser({name: serviceName, protocol: 'tcp'}),
-	_browserUp = false;
+var util = require('util'),
+		EventEmitter = require('events').EventEmitter,
+		conman = require('./con-man');
 
-module.exports = function(api_token) {
-	var self = this,
-		_apiToken = api_token,
-		_ws = null,
-		_queue = [],
-		_appsUp = false,
-		_logger = function(){};
+function ExceptionLogger() {
 
-	if (!_browserUp) {
-		_browserUp = true;
-		browser.start();
-	}
-
-	browser.on('serviceUp', function(service) {
-		if (_appsUp) return;
-		_appsUp = true;
-
-		_logger("[exception-logger] connecting to");
-		_logger(service);
-
-		if (_ws && _ws.readyState === 1) return;
-
-		// do some stuff to authenticate a socket
-		// note: this is REALLY basic and suck-ish
-		var _authenticate = function(socket, token) {
-			return new Promise(function(resolve, reject) {
-				socket.on('message', function(event) {
-					var p = JSON.parse(event.data);
-					_logger("[exception-logger] ", p);
-					if (p.type === "api_token-response") {
-						if (p.data === "OK") {
-							return resolve(p.data);
-						}
-					}
-				});
-				socket.send(JSON.stringify({type: "api_token", data: token}));
-
-				setTimeout(function() {
-					reject("TIMEOUT");
-				}, serviceTimeout);
-			});
-		};
-
-		// do some bs to make a socket that, on close, recreates itself
-		var createSocket = function(host, port) {
-			_logger("[exception-logger] creating socket["+host+":"+port+"]...");
+	var self = this;

```

Comment:
same as previously, it'd be better to just `.bind(this)` instead of assigning variables